### ISSUE TITLE
TypeError: can only concatenate str (not "NoneType") to str

### ISSUE THREAD

**Issue by martydingo** [mattieserver/netbox-topology-views#15]
Sorry to be a nuisance, I'm seeing this error, I've located it down to a switch that does bring fibre in, but it doesn't use any circuits. All devices connected are named, so unsure why they're being returned as a NoneType. 

Is there a way we can fallback to the device type name if there is no device name? And maybe the same for termination names? (It would fallback to the cable type and an index number to keep it unique until named?) 

```
Internal Server Error: /api/plugins/topology-views/search/search/
Traceback (most recent call last):
  File "/srv/netbox/venv/lib/python3.7/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/srv/netbox/venv/lib/python3.7/site-packages/django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/srv/netbox/venv/lib/python3.7/site-packages/django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/srv/netbox/venv/lib/python3.7/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "/srv/netbox/venv/lib/python3.7/site-packages/rest_framework/viewsets.py", line 114, in view
    return self.dispatch(request, *args, **kwargs)
  File "/srv/netbox/venv/lib/python3.7/site-packages/rest_framework/views.py", line 505, in dispatch
    response = self.handle_exception(exc)
  File "/srv/netbox/venv/lib/python3.7/site-packages/rest_framework/views.py", line 465, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/srv/netbox/venv/lib/python3.7/site-packages/rest_framework/views.py", line 476, in raise_uncaught_exception
    raise exc
  File "/srv/netbox/venv/lib/python3.7/site-packages/rest_framework/views.py", line 502, in dispatch
    response = handler(request, *args, **kwargs)
  File "/srv/netbox/venv/lib/python3.7/site-packages/netbox_topology_views/api/views.py", line 135, in search
    edge["title"] = "Connection between <br> " + cable.termination_a.device.name + " [" + cable.termination_a.name +  "]<br>" + cable.termination_b.device.name + " [" + cable.termination_b.name + "]"
TypeError: can only concatenate str (not "NoneType") to str
```

**Comment by martydingo** [comment]
I'm also seeing a different error for unnamed patch panels, I don't particularly want to figure out a naming convention for *every* patch panel we operate. 

```
Internal Server Error: /api/plugins/topology-views/search/search/
Traceback (most recent call last):
  File "/srv/netbox/venv/lib/python3.7/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/srv/netbox/venv/lib/python3.7/site-packages/django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/srv/netbox/venv/lib/python3.7/site-packages/django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/srv/netbox/venv/lib/python3.7/site-packages/django/views/decorators/csrf.py", line 54, in wrapped_view
    return view_func(*args, **kwargs)
  File "/srv/netbox/venv/lib/python3.7/site-packages/rest_framework/viewsets.py", line 114, in view
    return self.dispatch(request, *args, **kwargs)
  File "/srv/netbox/venv/lib/python3.7/site-packages/rest_framework/views.py", line 505, in dispatch
    response = self.handle_exception(exc)
  File "/srv/netbox/venv/lib/python3.7/site-packages/rest_framework/views.py", line 465, in handle_exception
    self.raise_uncaught_exception(exc)
  File "/srv/netbox/venv/lib/python3.7/site-packages/rest_framework/views.py", line 476, in raise_uncaught_exception
    raise exc
  File "/srv/netbox/venv/lib/python3.7/site-packages/rest_framework/views.py", line 502, in dispatch
    response = handler(request, *args, **kwargs)
  File "/srv/netbox/venv/lib/python3.7/site-packages/netbox_topology_views/api/views.py", line 143, in search
    node["label"] = device.name + " " + device.device_type.display_name
TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'
```

**Comment by mattieserver** [comment]
can you check if you still have this issue with the latest release?

**Comment by martydingo** [comment]
@mattieserver Nice one, thanks for the quick fix! I can confirm that this is working now.

**Comment by martydingo** [comment]
Just a slight issue, the word 'Unknown' is spelt wrong

**Comment by martydingo** [comment]
Aha, I thought I'd have a chance to beat you to the close. 

Sorry

**Comment by mattieserver** [comment]
should be fixed in v0.4.4

**Comment by martydingo** [comment]
Thanks for that!
